### DISCUSSION TITLE
Alias and variable template versions of Boost.Geometry traits

### DISCUSSION THREAD

**Discussion by awulkiew** [boostorg/geometry#1249] [Ideas]
What do you think about adding alias and variable templates to the library?

Besides being shorter than the original traits this would also allow us to avoid using `typename` keyword, so:
```
typename point_type<T>::type                      -> point_t<T>
typename select_calculation_type<G1, G2, T>::type -> select_calculation_t<G1, G2, T>
dimension<T>::value                               -> dimension_v<T>
is_multi<T>::value                                -> is_multi_v<T>
etc.
```

I haven't looked into this deeply but maybe we could unify `calculation_type`, `select_calculation_type` and `select_calculation_type_alt` as well?

**Comment by barendgehrels** [comment]
👍 yes, this is quite common now, and shorter. I like it, and also unifying calculation types.
It's all used throughout the code - changing it will be quite some work...
